### PR TITLE
Either.value with some updates

### PR DESCRIPTION
**Pull request by bvenners**

### CODE REVIEW

**Comment by bvenners** [comment]
@cheeseng can you check this with MIMA? Also I'd like to question the SerialVersionIds. Should we have those or would it be better to leave them off. For the time being I didn't add one for the new class I added (which I'm hoping won't break binary compatibility).

**Comment by cheeseng** [comment]
@bvenners I ran this through MIMA test, for scala 2.10 and 2.11 it fails with: 

```
[error] scalatest-core: Failed binary compatibility check against org.scalatest:scalatest-core_2.10:3.2.2! Found 1 potential problems
[error]  * method convertEitherToValuable(scala.util.Either,org.scalactic.source.Position)org.scalatest.EitherValues#Valuable in trait org.scalatest.EitherValues is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("org.scalatest.EitherValues.convertEitherToValuable")
[error] java.lang.RuntimeException: Failed binary compatibility check against org.scalatest:scalatest-core_2.10:3.2.2! Found 1 potential problems
[error]         at scala.sys.package$.error(package.scala:30)
[error]         at com.typesafe.tools.mima.plugin.SbtMima$.reportModuleErrors(SbtMima.scala:75)
[error]         at com.typesafe.tools.mima.plugin.MimaPlugin$.$anonfun$projectSettings$2(MimaPlugin.scala:38)
[error]         at com.typesafe.tools.mima.plugin.MimaPlugin$.$anonfun$projectSettings$2$adapted(MimaPlugin.scala:28)
[error]         at scala.collection.Iterator.foreach(Iterator.scala:941)
[error]         at scala.collection.Iterator.foreach$(Iterator.scala:941)
[error]         at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
[error]         at com.typesafe.tools.mima.plugin.MimaPlugin$.$anonfun$projectSettings$1(MimaPlugin.scala:28)
[error]         at com.typesafe.tools.mima.plugin.MimaPlugin$.$anonfun$projectSettings$1$adapted(MimaPlugin.scala:26)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:281)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error]         at sbt.Execute.work(Execute.scala:290)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:281)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]         at java.lang.Thread.run(Thread.java:748)
[error] (scalatestCore / mimaReportBinaryIssues) Failed binary compatibility check against org.scalatest:scalatest-core_2.10:3.2.2! Found 1 potential problems
[error] Total time: 17 s, completed Oct 2, 2020 10:54:45 AM
```

which I think should be fine, do you concur?  I'll submit a PR to ignore that error against your branch.

**Comment by cheeseng** [comment]
@bvenners I submitted the following PR for ignoring the related MIMA check: 

https://github.com/scalatest/scalatest/pull/1901

File: jvm/core/src/main/scala/org/scalatest/EitherValues.scala [hunk 1]
```diff
@@ -87,17 +88,26 @@ trait EitherValues extends Serializable {
...
+  @deprecated("The .right.value syntax on Either has been deprecated and will be removed in a future version of ScalaTest. Please use .value instead.")
   implicit def convertRightProjectionToValuable[L, R](rightProj: Either.RightProjection[L, R])(implicit pos: source.Position): RightValuable[L, R] = new RightValuable(rightProj, pos)
 
+  /**
+   * Implicit conversion that adds a <code>value</code> method to <code>Either</code>.
+   * This method is right biased and is the equivalent of calling <code>either.right.value</code>.
+   *
+   * @param either the <code>Either</code> on which to add the <code>value</code> method
+   */
+  implicit def convertEitherToValuable[L, R](either: Either[L, R])(implicit pos: source.Position): Valuable[L, R] = new Valuable(either, pos)
```
Comment (fancywriter) [inline jvm/core/src/main/scala/org/scalatest/EitherValues.scala:109]: why not to use implicit class instead of implicit conversion?

**Pull request merged**